kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -56,6 +56,39 @@ class RobotState;
 namespace kinematics
 {
 
+/*
+ * @enum KinematicSearches
+ * @brief Flags for choosing the type discretization applied on the redundancy during a multiple ik solution search
+ */
+namespace KinematicSearches
+{
+  enum KinematicSearch
+  {
+    ONE = 1,
+    ALL_DISCRETIZED ,
+    SOME_DISCRETIZED,
+    RANDOM_SAMPLE,
+    ALL
```

Comment:
This seems very specific to redundant joints. Its also not something most plugins may implement. Why not let this live in the plugins themselves - is there a case where you have to use one of these methods for some calls and then another one later on?